File: app/src/main/java/com/example/massa/luxvilla/Actividades/Userprofile.java
package com.example.massa.luxvilla.Actividades;

import android.content.Intent;
import android.net.Uri;
import android.support.annotation.NonNull;
import android.support.design.widget.CollapsingToolbarLayout;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.content.ContextCompat;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.example.massa.luxvilla.R;
import com.example.massa.luxvilla.separadores.separadorlikes;
import com.example.massa.luxvilla.separadores.separadorsobre;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.auth.UserInfo;
import com.squareup.picasso.Picasso;

import de.hdodenhof.circleimageview.CircleImageView;

public class Userprofile extends AppCompatActivity {
    CircleImageView profileimage;
    FirebaseAuth mAuth;
    FirebaseUser user;
    String profileDisplayName;
    Uri profilePhotoUrl;
    CollapsingToolbarLayout toolbarLayout;
    Toolbar toolbar;
    ViewPager vwpgr;
    TabLayout tbs;
    final int SEPARADOR_SOBRE=0,SEPARADOR_LIKES=1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_userprofile);

        toolbarLayout= findViewById(R.id.tbprofile);
        toolbar= findViewById(R.id.barprofileactivity);
        toolbar.setNavigationIcon(R.mipmap.ic_arrow_back_white_24dp);
        setSupportActionBar(toolbar);
        profileimage= findViewById(R.id.imgprofile);
        mAuth=FirebaseAuth.getInstance();
        user=mAuth.getCurrentUser();
        if (user !=null) {
            for (UserInfo userdata: user.getProviderData()) {

                profileDisplayName=userdata.getDisplayName();
                toolbarLayout.setTitle(profileDisplayName);
                profilePhotoUrl=userdata.getPhotoUrl();
            }
            if (profilePhotoUrl !=null){
                String image=profilePhotoUrl.toString();
                Picasso.with(Userprofile.this)
                        .load(image)
                        .fit()
                        .into(profileimage);
            }else{
                profileimage.setImageDrawable(ContextCompat.getDrawable(Userprofile.this,R.drawable.nouserimage));
            }
        }
        tbs= findViewById(R.id.profiletabs);
        tbs.setSelectedTabIndicatorColor(ContextCompat.getColor(Userprofile.this,R.color.colorAccent));
        vwpgr= findViewById(R.id.profilevpgr);
        adaptadortabs adaptador=new adaptadortabs(getSupportFragmentManager());
        vwpgr.setAdapter(adaptador);
        tbs.setupWithViewPager(vwpgr);

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_profile, menu);

        return true;
    }

    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()){
            case android.R.id.home:
                onBackPressed();
                break;
            case R.id.editprofile:
                startActivity(new Intent(Userprofile.this,Editprofile.class));
                break;
        }
        return super.onOptionsItemSelected(item);
    }
    public void onResume(){
        super.onResume();

        if (user !=null){
            user.reload().addOnSuccessListener(new OnSuccessListener<Void>() {
                @Override
                public void onSuccess(Void aVoid) {
                    FirebaseAuth auth=FirebaseAuth.getInstance();
                    FirebaseUser user=auth.getCurrentUser();
                    if (user !=null){
                        for (UserInfo userdata: user.getProviderData()) {
                            profileDisplayName=userdata.getDisplayName();
                            toolbarLayout.setTitle(profileDisplayName);
                            profilePhotoUrl=userdata.getPhotoUrl();

                        }
                        if (profilePhotoUrl !=null){
                            String image=profilePhotoUrl.toString();
                            Picasso.with(Userprofile.this)
                                    .load(image)
                                    .fit()
                                    .into(profileimage);
                        }else{
                            profileimage.setImageDrawable(ContextCompat.getDrawable(Userprofile.this,R.drawable.nouserimage));
                        }
                    }

                }
            }).addOnFailureListener(new OnFailureListener() {
                @Override
                public void onFailure(@NonNull Exception e) {

                }
            });
        }
    }

    private class adaptadortabs extends FragmentPagerAdapter{

        adaptadortabs(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            switch (position){
                case SEPARADOR_SOBRE:
                    return new separadorsobre();
                case SEPARADOR_LIKES:
                    return new separadorlikes();
            }
            return null;
        }

        public CharSequence getPageTitle(int position){
            String[] tabsname={"SOBRE","LIKES"};

            return tabsname[position];
        }

        @Override
        public int getCount() {
            return 2;
        }
    }
}

File: app/src/main/java/com/example/massa/luxvilla/Actividades/Userprofile.kt
package com.example.massa.luxvilla.Actividades

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import com.example.massa.luxvilla.R
import com.example.massa.luxvilla.separadores.separadorlikes
import com.example.massa.luxvilla.separadores.separadorsobre
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_userprofile.*

/**
 * Created by massa on 27/10/2017.
 */
class Userprofile : AppCompatActivity() {
    internal var mAuth: FirebaseAuth? = null
    internal var user: FirebaseUser? = null
    internal var profileDisplayName: String? = null
    internal var profilePhotoUrl: Uri? = null
    internal val SEPARADOR_SOBRE = 0
    internal val SEPARADOR_LIKES = 1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_userprofile)

        barprofileactivity.setNavigationIcon(R.mipmap.ic_arrow_back_white_24dp)
        setSupportActionBar(barprofileactivity)
        mAuth = FirebaseAuth.getInstance()
        user = mAuth!!.currentUser
        if (user != null) {
            for (userdata in user!!.providerData) {

                profileDisplayName = userdata.displayName
                tbprofile.title = profileDisplayName
                profilePhotoUrl = userdata.photoUrl
            }
            if (profilePhotoUrl != null) {
                val image = profilePhotoUrl!!.toString()
                Picasso.with(this@Userprofile)
                        .load(image)
                        .fit()
                        .into(imgprofile)
            } else {
                imgprofile.setImageDrawable(ContextCompat.getDrawable(this@Userprofile, R.drawable.nouserimage))
            }
        }
        profiletabs.setSelectedTabIndicatorColor(ContextCompat.getColor(this@Userprofile, R.color.colorAccent))
        val adaptador = adaptadortabs(supportFragmentManager)
        profilevpgr.adapter = adaptador
        profiletabs.setupWithViewPager(profilevpgr)

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_profile, menu)

        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> onBackPressed()
            R.id.editprofile -> startActivity(Intent(this@Userprofile, Editprofile::class.java))
        }
        return super.onOptionsItemSelected(item)
    }

    public override fun onResume() {
        super.onResume()

        if (user != null) {
            user!!.reload().addOnSuccessListener {
                val auth = FirebaseAuth.getInstance()
                val user = auth.currentUser
                if (user != null) {
                    for (userdata in user.providerData) {
                        profileDisplayName = userdata.displayName
                        tbprofile.title = profileDisplayName
                        profilePhotoUrl = userdata.photoUrl

                    }
                    if (profilePhotoUrl != null) {
                        val image = profilePhotoUrl!!.toString()
                        Picasso.with(this@Userprofile)
                                .load(image)
                                .fit()
                                .into(imgprofile)
                    } else {
                        imgprofile.setImageDrawable(ContextCompat.getDrawable(this@Userprofile, R.drawable.nouserimage))
                    }
                }
            }.addOnFailureListener { }
        }
    }

    private inner class adaptadortabs internal constructor(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        override fun getItem(position: Int): Fragment? {
            when (position) {
                SEPARADOR_SOBRE -> return separadorsobre()
                SEPARADOR_LIKES -> return separadorlikes()
            }
            return null
        }

        override fun getPageTitle(position: Int): CharSequence {
            val tabsname = arrayOf("SOBRE", "LIKES")

            return tabsname[position]
        }

        override fun getCount(): Int {
            return 2
        }
    }
}
